sql/core/src/test/scala/org/apache/spark/sql/SQLConfSuite.scala

```diff
@@ -73,8 +77,13 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
 
   test("deprecated property") {
     sqlContext.conf.clear()
-    sql(s"set ${SQLConf.Deprecated.MAPRED_REDUCE_TASKS}=10")
-    assert(sqlContext.conf.numShufflePartitions === 10)
+    val original = sqlContext.conf.numShufflePartitions
+    try{
+      sql(s"set ${SQLConf.Deprecated.MAPRED_REDUCE_TASKS}=10")
+      assert(ctx.conf.numShufflePartitions === 10)
```

Comment:
There is a merge conflict with https://github.com/apache/spark/pull/8584 here, we should change `ctx` to `sparkContext`. https://github.com/apache/spark/pull/8612 fixed it.